### PR TITLE
feat(console): OIDC setup

### PR DESCRIPTION
**Pull request by peintnermax**

### CODE REVIEW

File: console/src/app/pages/projects/apps/app-create/app-create.component.ts [hunk 1]
```diff
@@ -37,10 +46,18 @@ export class AppCreateComponent implements OnInit, OnDestroy {
...
+        NATIVE_TYPE,
+        USER_AGENT_TYPE,
+    ];
+
+    public authMethods: RadioItemAuthType[] = [
+        PKCE_METHOD,
+        CODE_METHOD,
+        POST_METHOD,
+        // PK_JWT_METHOD,
+        // IMPLICIT_METHOD,
```
Comment (livio-a) [inline console/src/app/pages/projects/apps/app-create/app-create.component.ts:60]: ?

File: console/src/app/pages/projects/apps/app-create/app-create.component.ts [hunk 2]
```diff
@@ -215,91 +231,42 @@ export class AppCreateComponent implements OnInit, OnDestroy {
...
-    changeResponseType(): void {
-        this.oidcApp.responseTypesList = this.oidcResponseTypes.filter(gt => gt.checked).map(gt => gt.type);
-    }
-
-    moreThanOneOption(options: Array<{ type: OIDCGrantType, checked: boolean, disabled: boolean; }>): boolean {
-        return options.filter(option => option.disabled === false).length > 1;
-    }
+    // moreThanOneOption(options: Array<{ type: OIDCGrantType, checked: boolean, disabled: boolean; }>): boolean {
+    //     return options.filter(option => option.disabled === false).length > 1;
+    // }
```
Comment (livio-a) [inline console/src/app/pages/projects/apps/app-create/app-create.component.ts:236]: ?

File: console/src/app/pages/projects/apps/app-detail/app-detail.component.ts [hunk 3]
```diff
@@ -176,6 +200,74 @@ export class AppDetailComponent implements OnInit, OnDestroy {
...
+            case OIDCApplicationType.OIDCAPPLICATIONTYPE_NATIVE:
+                this.authMethods = [
+                    PKCE_METHOD,
+                    CUSTOM_METHOD,
+                ];
+
+                // automatically set to PKCE and skip step
+                // this.app.oidcConfig.responseTypesList = [OIDCResponseType.OIDCRESPONSETYPE_CODE];
+                // this.app.oidcConfig.grantTypesList = [OIDCGrantType.OIDCGRANTTYPE_AUTHORIZATION_CODE];
+                // this.app.oidcConfig.authMethodType = OIDCAuthMethodType.OIDCAUTHMETHODTYPE_NONE;
```
Comment (livio-a) [inline console/src/app/pages/projects/apps/app-detail/app-detail.component.ts:214]: ?

File: console/src/assets/i18n/de.json [hunk 4]
```diff
@@ -991,13 +1013,58 @@
...
             "IDTOKENUSERINFOASSERTION_DESCRIPTION":"Ermöglich OIDC clients claims von profile, email, phone und address direkt vom ID Token zu beziehen.",
-            "CLOCKSKEW":"ermöglicht Clients, den Taktversatz von OP und Client zu verarbeiten. Die Dauer (0-5s) wird der exp addiert und von iats, auth_time und nbf abgezogen."
+            "CLOCKSKEW":"ermöglicht Clients, den Taktversatz von OP und Client zu verarbeiten. Die Dauer (0-5s) wird der exp addiert und von iats, auth_time und nbf abgezogen.",
+            "RECOMMENDED":"Empfohlen",
+            "NOTRECOMMENDED":"nicht empfohlen",
+            "SELECTION":{
+                "APPTYPE": {
+                    "WEB": {
+                        "TITLE":"Web",
+                        "DESCRIPTION":"Regular Web applications like .net, PHP, Node.js, Java, etc."
```
Comment (livio-a) [inline console/src/assets/i18n/de.json:1023]:
```suggestion
                        "DESCRIPTION":"Standard Web applications wie .net, PHP, Node.js, Java, etc."
```

Comment (livio-a) [reply console/src/assets/i18n/de.json:1023]: not sure

File: console/src/assets/i18n/de.json [hunk 5]
```diff
@@ -991,13 +1013,58 @@
...
+                    },
+                    "NATIVE": {
+                        "TITLE":"NATIVE",
+                        "DESCRIPTION":"Mobile Apps, Desktop, Smart Devices, etc."
+                    },
+                    "USERAGENT": {
+                        "TITLE":"User Agent",
+                        "DESCRIPTION":"Single Page Apüplications (SPA) and in general all js frameworks executed in browsers"
```
Comment (livio-a) [inline console/src/assets/i18n/de.json:1031]:
```suggestion
                        "DESCRIPTION":"Single Page Applications (SPA) und grundsätzlich alle im Browser aufgeführten JS Frameworks"
```

File: console/src/assets/i18n/de.json [hunk 6]
```diff
@@ -991,13 +1013,58 @@
...
+                    }
+                },
+                "AUTHMETHOD": {
+                    "CODE": {
+                        "TITLE":"Code",
+                        "DESCRIPTION":"Exchange the authorization code for the tokens"
```
Comment (livio-a) [inline console/src/assets/i18n/de.json:1037]:
```suggestion
                        "DESCRIPTION":"Tausche den Authorization Code gegen Tokens ein"
```

File: console/src/assets/i18n/de.json [hunk 7]
```diff
@@ -991,13 +1013,58 @@
...
+                    },
+                    "PKCE": {
+                        "TITLE":"PKCE",
+                        "DESCRIPTION":"Use a random hash instead of a static client secret for more security"
```
Comment (livio-a) [inline console/src/assets/i18n/de.json:1041]:
```suggestion
                        "DESCRIPTION":"Nutze einen Zufalls Hash Wert anstelle des Client Secret für mehr Sicherheit"
```

File: console/src/assets/i18n/de.json [hunk 8]
```diff
@@ -991,13 +1013,58 @@
...
+                    },
+                    "POST": {
+                        "TITLE":"POST",
+                        "DESCRIPTION":"Send client_id and client_secret as part of the form"
```
Comment (livio-a) [inline console/src/assets/i18n/de.json:1045]:
```suggestion
                        "DESCRIPTION":"Sende client_id und client_secret im (HTML) Formular"
```

File: console/src/assets/i18n/de.json [hunk 9]
```diff
@@ -991,13 +1013,58 @@
...
+                    },
+                    "PK_JWT": {
+                        "TITLE":"Private Key JWT",
+                        "DESCRIPTION":"Use a private key to authorize your application"
```
Comment (livio-a) [inline console/src/assets/i18n/de.json:1049]:
```suggestion
                        "DESCRIPTION":"Nutze einen Private Key um deine Application zu authentifizieren"
```

File: console/src/assets/i18n/de.json [hunk 10]
```diff
@@ -991,13 +1013,58 @@
...
+                    },
+                    "IMPLICIT": {
+                        "TITLE":"Implicit",
+                        "DESCRIPTION":"Get the tokens directly from the authorization endpoint"
```
Comment (livio-a) [inline console/src/assets/i18n/de.json:1053]:
```suggestion
                        "DESCRIPTION":"Erhalte die Token direkt vom authorize Endpoint"
```

File: console/src/assets/i18n/de.json [hunk 11]
```diff
@@ -991,13 +1013,58 @@
...
+                    },
+                    "CUSTOM": {
+                        "TITLE":"Custom",
+                        "DESCRIPTION":"Your setting doesn't correspond to any other option."
```
Comment (livio-a) [inline console/src/assets/i18n/de.json:1057]:
```suggestion
                        "DESCRIPTION":"Deine Konfiguration entspricht keiner anderen Option."
```

File: console/src/assets/i18n/en.json [hunk 12]
```diff
@@ -991,13 +1013,58 @@
...
+                        "TITLE":"Web",
+                        "DESCRIPTION":"Regular Web applications like .net, PHP, Node.js, Java, etc."
+                    },
+                    "NATIVE": {
+                        "TITLE":"NATIVE",
+                        "DESCRIPTION":"Mobile Apps, Desktop, Smart Devices, etc."
+                    },
+                    "USERAGENT": {
+                        "TITLE":"User Agent",
+                        "DESCRIPTION":"Single Page Apüplications (SPA) and in general all js frameworks executed in browsers"
```
Comment (livio-a) [inline console/src/assets/i18n/en.json:1031]:
```suggestion
                        "DESCRIPTION":"Single Page Applications (SPA) and in general all JS frameworks executed in browsers"
```

**Review by livio-a — COMMENTED** [review]
not sure about german translations... maybe you can think of better ones

File: console/src/app/pages/projects/apps/app-detail/app-detail.component.ts (hunk 3 above)
Comment (peintnermax) [reply console/src/app/pages/projects/apps/app-detail/app-detail.component.ts:214]: removed

File: console/src/app/pages/projects/apps/app-create/app-create.component.ts (hunk 2 above)
Comment (peintnermax) [reply console/src/app/pages/projects/apps/app-create/app-create.component.ts:236]: removed comment

File: console/src/app/pages/projects/apps/app-create/app-create.component.ts (hunk 1 above)
Comment (peintnermax) [reply console/src/app/pages/projects/apps/app-create/app-create.component.ts:60]: removed comment

**Review by livio-a — APPROVED** [review]

**Pull request merged**